src/Prefetch.cpp

```diff
@@ -0,0 +1,348 @@
+
+        pstmts.push_back(stmt_prefetch);
+    }
+
+    // Inject the generated prefetch code
+    //
+    // Note: plets/pstmts may be modified or partially emptied during injection
+    //
+    void inject_prefetch_stmts(const For *op, const Prefetch &p,
+                vector<pair<string, Expr>> &plets, vector<Stmt> &pstmts, Stmt &body)
+    {
+        if (pstmts.size() > 0) {
+            Stmt pbody = pstmts.back();    // Initialize prefetch body
+            pstmts.pop_back();
+            for (size_t i = pstmts.size(); i > 0; i--) {
+                pbody = Block::make(pstmts[i-1], pbody);
+            }
+            for (size_t i = plets.size(); i > 0; i--) {
+                pbody = LetStmt::make(plets[i-1].first, plets[i-1].second, pbody);
+            }
+
+            // No guard needed, address range checked in prefetch runtime
+            body = Block::make({pbody, body});
+            debug(dbg_prefetch4) << pbody << "\n";
+        }
+    }
+
+    void visit(const Let *op) {
+        Interval in = bounds_of_expr_in_scope(op->value, scope);
+        scope.push(op->name, in);
+        debug(dbg_prefetch5) << "Let scope.push(" << op->name << ")\n";
+        IRMutator::visit(op);
+        debug(dbg_prefetch5) << "Let scope.pop(" << op->name << ")\n";
+        scope.pop(op->name);
+    }
+
+    void visit(const LetStmt *op) {
+        Interval in = bounds_of_expr_in_scope(op->value, scope);
+        scope.push(op->name, in);
+        debug(dbg_prefetch5) << "LetStmt scope.push(" << op->name << ")\n";
+        IRMutator::visit(op);
+        debug(dbg_prefetch5) << "LetStmt scope.pop(" << op->name << ")\n";
+        scope.pop(op->name);
+    }
+
+    void visit(const Realize *op) {
+        rscope.push(op->name, 1);
+        debug(dbg_prefetch5) << "Realize push(" << op->name << ")\n";
+        IRMutator::visit(op);
+    }
+
+    void visit(const For *op) {
+        Stmt body = op->body;
+
+        string func_name = tuple_func(op->name);
+        string ivar_name = tuple_var(op->name);
+        vector<Prefetch> &prefetches = get_func(func_name).schedule().prefetches();
+
+        // Add loop variable to interval scope for any inner loop prefetch
+        Expr var = Variable::make(Int(32), op->name);
+        Interval prein(var, var);
+        scope.push(op->name, prein);
+        debug(dbg_prefetch5) << "For scope.push(" << op->name << ", (" << var << "))\n";
+
+        debug(dbg_prefetch4) << "For: " << op->name << " " << func_name << " " << ivar_name << "\n";
+        if (prefetches.empty()) {
+            debug(dbg_prefetch4) << " No prefetch directives in schedule\n";
+        } else {
+            debug(dbg_prefetch4) << " Found prefetch directive(s) in schedule\n";
+        }
+
+        body = mutate(body);
+
+        for (const Prefetch &p : prefetches) {
+            debug(dbg_prefetch4) << " Check prefetch ivar: " << p.var << " == " << ivar_name << "\n";
+            if (p.var == ivar_name) {
```

Comment:
This is the entire body of the loop, in this case, I'd suggest:

    if (p.var != ivar_name) continue;

And then just let the rest of the code be the loop body.